pkg/test/framework/components/zipkin/zipkin.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zipkin
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+// Instance represents a zipkin deployment on kube
+type Instance interface {
+	resource.Resource
+
+	// QueryTraces returns all available traces via zipkin query api
+	QueryTraces() ([]Trace, error)
```

Comment:
Does the Zipkin API allow filtering, do reduce the overhead of the request? If so, should we support that here?

Reply:
This code actually just retrieve 100 traces: https://github.com/istio/istio/pull/13790/files#diff-78d045de8858766f41109e4a83f6be8eR36. I updated the comment to reflect that.

Reply:
Ok, but what does zipkin support? Does it have more flexible filtering options ... and if so, should we make this API more flexible?

Also, 100 seems like a magic number .... should it be configurable?